### ISSUE TITLE
Adding the ability to freeze training of some layers

### ISSUE THREAD

**Issue by Joshuaalbert** [deeplearning4j/deeplearning4j#2884]
Having the ability to fix the weights of some layers/vertices in a network during training would allow more sophisticated fine tuning of networks, as well as align with the Keras API. I'm guessing the easiest way is to throw a `trainable` boolean into the base layer and graph vertex classes, and then the updater can decide whether to zero out a particular gradient or not (or equivalently set the learning rate to zero for layers with trainable = true).

**Comment by turambar** [comment]
I believe this is being addressed in some WIP transfer learning functionality that we're adding. Inviting @eraly @AlexDBlack to comment further.

**Comment by turambar** [comment]
The comment about the Keras API is a good one. @eraly how did we end up addressing frozen layers -- with a property in the layer itself or with a wrapper? I think we were leaning toward the latter, and I'm now wishing we'd chosen the former, which would be closer to Keras.

**Comment by eraly** [comment]
This will be part of the next release. We decided on a wrapper because it was the cleanest way to do it considering the complexities in our code base.

**Comment by turambar** [comment]
@agibsonccc looks like it's merged.

**Comment by eraly** [comment]
Ya, merged. Closing.
